Autograd gradcheck code clean up

anjali411 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #43208 [WIP] Complex gradcheck logic
* **#43167 Autograd gradcheck code clean up**

Differential Revision: [D23222358](https://our.internmc.facebook.com/intern/diff/D23222358)

anjali411 committed:
Autograd code clean up

[ghstack-poisoned]

anjali411 committed:
Update on "Autograd code clean up"




[ghstack-poisoned]

anjali411 committed:
Update on "Autograd code clean up"




[ghstack-poisoned]

anjali411 committed:
Update on "Autograd code clean up"




[ghstack-poisoned]

albanD commented on torch/autograd/gradcheck.py:
```diff
@@ -61,6 +61,24 @@ def get_numerical_jacobian(fn, input, target=None, eps=1e-3):
 
+    def compute_gradient(x, idx, is_mkldnn=False):
+
+        def fn_out(is_mkldnn=False):
```
Why do you need to pass this as an argument? You can just capture it from above?

anjali411 replied:
agreed will remove it

albanD commented on torch/autograd/gradcheck.py:
```diff
@@ -61,6 +61,24 @@ def get_numerical_jacobian(fn, input, target=None, eps=1e-3):
+
+        def fn_out(is_mkldnn=False):
+            if not is_mkldnn:
+                return fn(input).clone()
```
Is that expected that you just capture `input` from the parent scope here?

anjali411 replied:
yeah

albanD replied:
You might want to add a comment here stating that x is a view into input and so it just works.

albanD commented on torch/autograd/gradcheck.py:
```diff
@@ -61,6 +61,24 @@ def get_numerical_jacobian(fn, input, target=None, eps=1e-3):
+        outa = fn_out(is_mkldnn)
+        x[idx] = orig + eps
+        outb = fn_out(is_mkldnn)
+        if not is_mkldnn:
```
I think this is fairly confusing if you don't see the rest of the code from the mkldnn part. Because it would suggest we don't write back the original value.

anjali411 replied:
I think it's also ok if we remove this check. we don't have to set back the orig value for x in case of mkldnn since we are not overwriting the x_tensor in the code below but it's also ok to set it back.

albanD reviewed (commented):
The overall idea is nice.
Maybe this would need a few comments to make it clearer because you pull far away things that used to be next to each other in the code.

anjali411 committed:
Update on "Autograd code clean up"




[ghstack-poisoned]

anjali411 committed:
Update on "Autograd code clean up"




[ghstack-poisoned]

anjali411 committed:
Update on "Autograd gradcheck code clean up"


Differential Revision: [D23222358](https://our.internmc.facebook.com/intern/diff/D23222358)

[ghstack-poisoned]

anjali411 committed:
Update on "Autograd gradcheck code clean up"


Differential Revision: [D23222358](https://our.internmc.facebook.com/intern/diff/D23222358)

[ghstack-poisoned]

anjali411 committed:
Update on "Autograd gradcheck code clean up"


Differential Revision: [D23222358](https://our.internmc.facebook.com/intern/diff/D23222358)

[ghstack-poisoned]
